File: spcserver/springboot/src/main/java/com/lcaohoanq/shoppe/domain/token/ITokenService.java
package com.lcaohoanq.shoppe.domain.token;

import com.lcaohoanq.shoppe.base.exception.DataNotFoundException;
import com.lcaohoanq.shoppe.domain.user.User;
import org.springframework.stereotype.Service;

@Service

public interface ITokenService {
    Token addToken(long userId, String token, boolean isMobileDevice);
    Token refreshToken(String refreshToken, User user) throws Exception;
    void deleteToken(String token, User user) throws DataNotFoundException;
    Token findUserByToken(String token) throws DataNotFoundException;
}

File: kt-service/src/main/kotlin/com/lcaohoanq/ktservice/domain/token/ITokenService.kt
package com.lcaohoanq.ktservice.domain.token

import com.lcaohoanq.ktservice.entities.Token
import com.lcaohoanq.ktservice.entities.User


interface ITokenService {

    fun addToken(userId: Long, token: String): Token
    fun refreshToken(refreshToken: String, user: User): Token
    fun deleteToken(token: String, user: User)
    fun findUserByToken(token: String): Token

}